### ISSUE TITLE
TSC/TSR: orchestration/benchmarking framework

### ISSUE THREAD

**Issue by fkiraly** [alan-turing-institute/sktime#11]
Worfklow automation, method evaluation, benchmarking and post-hoc analyses
Should be straightforward (but slightly tedious) by adapting mlaut framework interface

Conditional on evaluation framework https://github.com/kiraly-group/sktime/issues/3 which in turn is conditional on prediction interface https://github.com/kiraly-group/sktime/issues/2

**Comment by ViktorKaz** [comment]
A version of the orchestration framework is implemented in the `orchestration` branch.
It can handle both in memory data and loading the datasets from the HDD.
The results are written in the uae format using `from sktime.utils.results_writing.write_results_to_uea_format` but this can be easily changed if another format is deemed more appropriate.

If you want to perform the bake off you need to save all datasets in separate directories and split them in test and train folds. I was working off the format of the datasets saved here: https://drive.google.com/open?id=1gWnOD7pRkHcDHDn8CSp9IH66DYychOxa

Please see below an example of how this can be done:
```python
from sktime.experiments.data import DataHolder
from sktime.classifiers.ensemble import TimeSeriesForestClassifier
from sktime.highlevel import Task, TSCStrategy

clf = TimeSeriesForestClassifier()
strategy = TSCStrategy(clf)

dl = DataLoader(dts_dir='path/to/root/directory/where/dts/are/saved', task_types='TSC')
results = orchestrator.fit(data=dl,
                            strategies=[strategy]) #you can provide an array of strategies here, not just one
```

The orchestrator iterates efficiently though the datasets by loading only one dataset in memory at any point in time.

The `results` object returned from the estimator can be used for evaluation and benchmarking.

For example:

```Python
from sktime.analyze_results import AnalyseResults
from sktime.analyze_results.scores import ScoreAccuracy

analyze = AnalyseResults(results)
losses, losses_df = analyze.prediction_errors(metric=ScoreAccuracy())

avg_and_std = analyze.average_and_std_error(losses)
#other tests and functions are also implemented check sktime.analyze_results
#for example
analyze.plot_boxcharts(losses)
analyze.ranks(losses)
analyze.t_test(losses)
analyze.sign_test(losses)
analyze.ranksum_test(losses)
analyze.t_test_with_bonferroni_correction(losses)
analyze.wilcoxon_test(losses)
analyze.friedman_test(losses)
analyze.nemenyi(losses)

#This is copy-paste from mlaut and not all of these are relevant in the time series context
```
Further development needed:
1. Need to make the framework more general. At the moment it can handle only two use cases - in-memory and from HDD and relies on a certain structure of the datasets. If it is violated, the code will break. For example if the test/train splits are not saved in the directory with the correct suffix or if the `sktime.utils.load_data.load_from_tsfile_to_dataframe` function returns only the X without the y for a particular dataset. Do we have datasets with no class labels?
2. The orchestrator should save the estimators. However, this should probably be implemented in the strategy class, not the orchestrator.
3. The experiments should resume from the last trained strategy in case of a crash.
4. Implement the option to recreate the `results` object from the saved results of the experiments.
5. Expand the available statistical tests in `analyze_results` to include metrics relevant for the time series context such as Diebold-Mariano test, etc.

**Comment by ViktorKaz** [comment]
I wrote another implementation that handles the same use case in which the datasets are saved on disk and are pre-split. 

The difference is that:
1. the tasks are specified manually, and not created by the orchestrator
2. the paths to the datasets are individually specified.
3. the orchestrator design also potentially allows k-fold cross validation. Check the new orchestrator.run() method. However, we this is not fully implemented at the moment. We need to discuss how the predictions on the k splits need to be handled (calculate average, most votes, etc.)

From the user perspective this looks like this:

```python
from sktime.experiments.orchestrator import Orchestrator
from sktime.highlevel import Task, TSCStrategy
from sktime.classifiers.ensemble import TimeSeriesForestClassifier
from sktime.experiments.data import DatasetHDD
from sktime.model_selection import SKtime_PredefinedSingleSplit
from sktime.datasets import load_gunpoint
import pandas as pd


dts_ArrowHead = DatasetHDD(dataset_loc='data/datasets/ArrowHead', dataset_name='ArrowHead', target='target')
task_ArrowHead = Task(case='TSC', data=dts_ArrowHead, target='target')

dts_Beef = DatasetHDD(dataset_loc='data/datasets/Beef', dataset_name='Beef')
task_Beef = Task(case='TSC', data=dts_Beef, target='target')

clf = TimeSeriesForestClassifier()
strategy = TSCStrategy(clf)

orchestrator = Orchestrator()

predef_split = SKtime_PredefinedSingleSplit(dataset_loc='data/datasets', train_suffix='_TRAIN.ts', test_suffix='_TEST.ts')

orchestrator.run(tasks=[task_ArrowHead, task_Beef], datasets=[dts_ArrowHead, dts_Beef], strategies=[strategy], cv=predef_split)
```

**Comment by mloning** [comment]
Closed by #51
